src/status_im/ui/screens/chat/message/link_preview.cljs

```diff
@@ -88,16 +98,58 @@
                           :style styles/link-preview-site}
                 site]])]])))))
 
+(defview community-preview [community outgoing timeline]
+  (let [{:keys [name members description verified]} community
+        members-count (count members)]
+    [react/view (styles/link-preview-wrapper outgoing timeline)
+     (if verified [quo/text {:size :small
+                             :color :link
+                             :style styles/community-preview-header}
+                   (i18n/label :t/verified-community)]
+         [quo/text {:size :small
+                    :color :secondary
+                    :style styles/community-preview-header}
+          (i18n/label :t/community)])
+     [quo/separator]
+     [react/view {:flex-direction :row :align-self :flex-start :margin 12}
+      [communities.icon/community-icon community]
+      [react/view {:flex 1 :flex-direction :column :margin-left 12}
+       [quo/text {:weight :bold :size :large} name]
+       [quo/text description]
+       [quo/text {:size :small
+                  :color :secondary}
+        (i18n/label-pluralize members-count :t/community-members {:count members-count})]]]
+     [quo/separator]
+     [quo/button {:on-press #(re-frame/dispatch [:navigate-to
+                                                 :community
+                                                 {:community-id (:id community)}])
+                  :type     :secondary}
+      (i18n/label :t/view)]]))
+
+(defview community-preview-loader [community-link outgoing timeline]
+  (letsubs [cache [:link-preview/cache]]
+    {:component-did-mount (fn []
+                            (let [community (get cache community-link)
+                                  community-id (community-id-from-link community-link)]
+                              (when-not community
+                                (re-frame/dispatch
+                                 [::link-preview/resolve-community-info community-id]))))}
+    (let [community (get cache community-link)]
+      (when community
+        [community-preview community outgoing timeline]))))
+
 (defview link-preview-wrapper [links outgoing timeline]
   (letsubs
     [ask-user? [:link-preview/link-preview-request-enabled]
      whitelist [:link-preview/whitelist]
      enabled-sites   [:link-preview/enabled-sites]]
     (when links
       (let [link-info (previewable-link links whitelist enabled-sites)
-            {:keys [link whitelisted enabled]} link-info]
-        (when (and link whitelisted)
-          (if enabled
-            [link-preview-loader link outgoing timeline]
-            (when ask-user?
-              [link-preview-enable-request])))))))
+            {:keys [link whitelisted enabled community]} link-info]
+        (if community
+          [community-preview-loader link outgoing timeline]
+          (when (and link whitelisted)
```

Comment:
maybe this can be simplified a bit by using `cond`
```
(cond 
  community [community-pr..]
  (and link whitelisted)
  (cond 
     enabled [link-prev..
      ask-user? [...]))
```

or you can flatten it by `letting`
```
(let [link-whitelisted (and link-whitelisted)]
  (cond 
    community [community-preview...]
    (and link-whitelisted enabled) [link-preview-loader]
    (and link-whitelisted ask-user?) [link-preview-enable-request])
```